### PR TITLE
Make sure lazy seqs get realized during CSV exports

### LINKED ISSUE #26261: BigQuery Arrays are not exported correctly to CSV
**Describe the bug**
When exporting results of a Question containing BigQuery array columns to csv, the values of all array columns are replaced with the string `clojure.Lang.LazySeq@MEM_ADDR` where `MEM_ADDR` is the address of the Seq.

The lazy collection appears to never be materialised. Observe that JSON and Excel exports export the correct values.

**Logs**

There is no error in the logs and the API call to download CSV returns correctly.

```
2022-11-07 09:02:32,115 DEBUG middleware.log :: POST /api/card/3211/query/csv 202 [ASYNC: completed] 8.5 s (30 DB calls) App DB connections: 2/15 Jetty threads: 2/50 (6 idle, 0 queued) (114 total active threads) Queries in flight: 0 (0 queued)
```

**To Reproduce**
1. Create a new SQL Question via database connected to BigQuery.
2. Use the following query to generate an array column:
```
SELECT ["ARRAY!"] AS ArrCol
UNION ALL
SELECT ["ARRAY! AGAIN", "YES"]
```
4. Click Download as CSV.
5. Observe the column contents being incorrect.

**Expected behavior**
CSV exports BigQuery arrays correctly like the other two options.

**Information about your Metabase Installation:**

Metabase 1.44.5 deployed via Helm with a Postgres backend database and BigQuery connections.

**Severity**

Low severity since the Excel export works well.

### PR DESCRIPTION
**Pull request by noahmoss**
Resolves #26261

The issue is that apparently BigQuery arrays are represented as Clojure lazy seqs when they get to the CSV streaming code, and the CSV writer just converts these to the string representation of the class (unlike XLSX/JSON which apparently realize the data automatically).

I've extended the `FormatValue` protocol to handle `clojure.lang.LazySeq` and convert it to a normal, non-lazy list before we write it.

### CODE REVIEW

File: test/metabase/query_processor/streaming/csv_test.clj [hunk 1]
```diff
@@ -55,3 +57,23 @@
...
+      (doall (map-indexed
+              (fn [i row] (qp.si/write-row! results-writer row i [] {}))
+              rows))
+      (qp.si/finish! results-writer {:row_count (count rows)}))
+    (let [bytea (.toByteArray bos)]
+      (rest (csv/read-csv (String. bytea))))))
+
+(deftest lazy-seq-realized-test
+  (testing "Lazy seqs within rows are automatically realized during exports (#26261)"
+    (let [row (first (csv-export [[(lazy-seq [1 2 3])]]))]
```
Comment (metamben) [inline test/metabase/query_processor/streaming/csv_test.clj:77]: Is testing the case when a `ZonedDateTime` is an element of the lazy list interesting?

**Review by metamben — APPROVED** [review]
LGTM

File: src/metabase/query_processor/streaming/common.clj [hunk 2]
```diff
@@ -29,6 +30,10 @@
   Object
   (format-value [this] this)
 
+  ISeq
+  (format-value [this]
+    (into [] (map format-value this)))
```
Comment (metamben) [inline src/metabase/query_processor/streaming/common.clj:35]: Out of curiosity, why do you prefer `(into [] (map format-value this))` to `(mapv format-value this)`? Also, if we keep `into`, I would write `(into [] (map format-value) this)`.

**Review by metamben — APPROVED** [review]

File: src/metabase/query_processor/streaming/common.clj (hunk 2 above)
Comment (noahmoss) [reply src/metabase/query_processor/streaming/common.clj:35]: no preference, forgot about `mapv`

**Pull request merged**